### PR TITLE
Bug 1918524: Update Makefile to work with containers

### PR DESCRIPTION
**Pull request by michaelgugino**

### CODE REVIEW

**Comment by elmiko** [comment]
i was able to run this, and it did complete through the unit tests. some of the tests pass, but some fail and seem to be missing `/usr/local/kubebuilder/bin/etcd`. the failures seem mixed between mao and vsphere tests for me.

seems like we might want to add separate make targets for the mao unit test and vsphere tests at some point too, just so that we can test with more granularity where needed.

**Comment by michaelgugino** [comment]
> i was able to run this, and it did complete through the unit tests. some of the tests pass, but some fail and seem to be missing `/usr/local/kubebuilder/bin/etcd`. the failures seem mixed between mao and vsphere tests for me.
> 
> seems like we might want to add separate make targets for the mao unit test and vsphere tests at some point too, just so that we can test with more granularity where needed.

We actually have to run 'make test' as that downloads kubebuilder into the container.  Make unit is only useful if using NO_DOCKER=1 and you have kubebuilder already installed.

This is somewhat confusing and we could probably change it, but this is the first step.

**Comment by elmiko** [comment]
ack, thanks for the clarification!

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,8 @@ ifeq ($(NO_DOCKER), 1)
   DOCKER_CMD =
   IMAGE_BUILD_CMD = imagebuilder
 else
-  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator golang:1.13
+  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator golang:1.15
+#  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6
```
Comment (elmiko) [inline Makefile:28]: are you going to leave this in for reference to the rhel stuff?

**Review by elmiko — APPROVED** [review]
this worked for me, did you have some follow on changes you wanted to add @michaelgugino ?
/approve

File: Makefile (hunk 1 above)
Comment (michaelgugino) [reply Makefile:28]: Yes, for now.  If you want to test with the same image we build with, comment/uncomment those lines.  Possibly in the future make it more configurable, but it want people that don't have a CI registry pull secret to be able to build and test our bits.

**Comment by JoelSpeed** [comment]
Seems to work on mac too
/lgtm

**Pull request merged**